### DISCUSSION TITLE
Migrating Data

### DISCUSSION THREAD

**Discussion by Wgreen92** [alexjustesen/speedtest-tracker#1074] [Q&A]
Hey there, I know a official update is coming to help with migrate from henrywhitaker3/speedtest-tracker to here but I thought id give it a shot through export from there and manual mysql db update. 

I have a couple questions someone might be able to help with.

1. How are speeds represented in the db. Actual speed Download: 853.39 Mbps. In db, Download: 106673409. What is this number / conversion?
2. How is "data" field generated in db. Its all pretty obvious data, but it looks like its just copied from somewhere on speedtest.net. I have previous scans result links, if I can find where this is copied from I can build a scraper to pull this data from previous results.
Example data: 
`"{\"type\":\"result\",\"timestamp\":\"2024-01-26T20:13:27Z\",\"ping\":{\"jitter\":0.582,\"latency\":3.019,\"low\":2.192,\"high\":3.400},\"download\":{\"bandwidth\":106673409,\"bytes\":1252910352,\"elapsed\":13202,\"latency\":{\"iqm\":25.808,\"low\":2.212,\"high\":261.758,\"jitter\":6.196}},\"upload\":{\"bandwidth\":117289004,\"bytes\":829928780,\"elapsed\":7106,\"latency\":{\"iqm\":7.920,\"low\":3.792,\"high\":228.850,\"jitter\":10.083}},\"packetLoss\":0,\"isp\":\"Fiber\",\"interface\":{\"internalIp\":\"172.17.0.14\",\"name\":\"eth0\",\"macAddr\":\"02:42:AC:11:00:0E\",\"isVpn\":false,\"externalIp\":\"xx.xx.xx.xxx\"},\"server\":{\"id\":xxxx,\"host\":\"speedtest.net\",\"port\":8080,\"name\":\"Network\",\"location\":\"Place\",\"country\":\"United States\",\"ip\":\"ip\"},\"result\":{\"id\":\"guid\",\"url\":\"link\",\"persisted\":true}}\n"`

Any help would be appreciated :)

**Comment by alexjustesen** [comment]
I wouldn't touch this yet... `0.15.x` is going to fix a data formatting issues, all those `\` shouldn't exist and was a bug I didn't catch until after I released it.

Import and Export are coming shortly after that fix as the devs of the admin panel did the work for me:

- Import: https://filamentphp.com/docs/3.x/actions/prebuilt-actions/import
- Export: https://filamentphp.com/docs/3.x/actions/prebuilt-actions/export
